pyomo/duality/lagrangian_dual.py

```diff
@@ -89,7 +89,7 @@ def _create_using(self, model, **kwds):
 
         # Walk constaints
         for (con_name, con_array) in sf.component_map(Constraint, active=True).items():
-            for con in (con_array[ndx] for ndx in con_array._index):
+            for con in (con_array[ndx] for ndx in con_array._index_set):
```

Comment:
Shouldn't this (and the subsequent 3 instances) all use the public `index_set()` method?